### ISSUE TITLE
Review: MoreEnumerable.PairUp()

### ISSUE THREAD

**Issue by GoogleCodeExporter** [blyzer/morelinq#42]
```
Purpose of code changes on this branch:

Rev 127: Implementation of the PairUp extension method, which combines
adjacent pairs in the source sequence and returns the result of applying a
user-defined projection to them. PairUp can operate on either odd-length or
even-length sequences. In the case of add-length sequences, an optional
imbalance strategy can be supplied, which defines whether PairUp should
skip the last (unpaired) value, or process the terminal element with a
default value pairing. The result of PairUp applied to a non-empty
sequence, will always be a sequence of half the length of the source sequence.

When reviewing my code changes, please focus on:

* The public interface of the extension method.
* The effectiveness and clarity of the available XML comment documentation.
* How well this operator fits into the MoreLINQ ecosystem of
extension methods.
* The extent of coverage available from the corresponding unit tests.
* Whether edge cases are correctly identified and handled in the
implementation and tests.

After the review, I'll merge this branch into:
/trunk

```

Original issue reported on code.google.com by `ambientl...@gmail.com` on 18 Jan 2010 at 5:08

**Comment by GoogleCodeExporter** [comment]
```
Set review issue status to: Started

```

Original comment by `azizatif` on 23 Jan 2010 at 11:20
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
Now migrated to Hg and available in clone:
http://code.google.com/r/azizatif-morelinq-evenmore/
This clone is ready to be cloned :O) for further review. It also has the 
main/default branch already merged in to bring it up to date.
```

Original comment by `azizatif` on 25 May 2012 at 11:38

**Comment by GoogleCodeExporter** [comment]
```
Following no response from ambientlion (who possibly/assuming lost interest 
given it has taken years to get this far), taking ownership of issue to 
consider as addition for MoreLINQ 2.0 milestone.
```

Original comment by `azizatif` on 12 Jun 2013 at 10:10
- Added labels: **Component-Logic**, **Milestone-Release2.0**, **Priority-High**, **Type-Enhancement**
- Removed labels: **Priority-Medium**, **Type-Review**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision ecbb044b3f86.
```

Original comment by `azizatif` on 14 Jun 2013 at 5:18
- Changed state: **Fixed**
